### PR TITLE
Change rise cache watchdog to kill existing process when restarting

### PR DESCRIPTION
**Pull request by andrecardoso**

### CODE REVIEW

**Comment by andrecardoso** [comment]
[(3) Defect 716](https://trello.com/c/jtDp51UL/4301-3-defect-716)

File: src/main/player/rise-cache-watchdog.js [hunk 1]
```diff
@@ -57,6 +57,13 @@ module.exports = {
   launchCache() {
     startCache();
   },
+  restartCache() {
+    if (cache) {
+      log.external('killing rise cache');
+      cache.kill('SIGINT');
```
Comment (santiagonoguez) [inline src/main/player/rise-cache-watchdog.js]: just curious, does SIGINT works also on windows ? There's also a quitChache function below, what's the difference, and shouldn't it be used ?

Comment (andrecardoso) [reply src/main/player/rise-cache-watchdog.js]: Those are good points. I'll test it on windows and try using the quiteCache function.

Comment (stulees) [reply src/main/player/rise-cache-watchdog.js]: @andrecardoso SIGINT will not work on Windows. Back in pre-modularized player, the `quitCache()` function was used but I think somewhere in the separation/refactoring of Launcher and player-electron, it was overlooked.

I think you should try the following only:

```
isCacheRunning()
      .catch((err)=>{
        log.external("restarting cache", inspect(err));
        module.exports.quitCache();
        module.exports.launchCache();
      });
  }, module.exports.getCheckInterval());
```

**Comment by andrecardoso** [comment]
I changed it to use `quitCache` and tested on both Windows and Linux and it is working. @stulees @santiagonoguez @tejohnso could you please review it again? Thanks.

**Pull request merged**